data/transactions/logic/eval.go

```diff
@@ -3533,7 +3589,7 @@ func opAppLocalPut(cx *EvalContext) {
 		return
 	}
 
-	addr, accountIdx, err := cx.accountReference(cx.stack[pprev])
+	addr, accountIdx, err := cx.mutableAccountReference(cx.stack[pprev])
```

Comment:
New method introduced so that we do not allow AVM code that will modify an account to use the addresses created in the group.  It's not that we *want* to prevent modification, but we have no way to encode it. So we must disallow.